### PR TITLE
fix aws_kms_public_key.public_key format - must be base64

### LINKED ISSUE #20595: `aws_kms_public_key.public_key` returning malformed data
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```
Terraform v1.0.0
on darwin_amd64
+ provider registry.terraform.io/hashicorp/aws v3.54.0
```

### Affected Resource(s)

* aws_kms_public_key

### Terraform Configuration Files

```hcl
resource "aws_kms_key" "key" {
  description              = "My Key"
  deletion_window_in_days  = 30
  key_usage                = "SIGN_VERIFY"
  customer_master_key_spec = "RSA_2048"
}

data "aws_kms_public_key" "key" {
  depends_on = [aws_kms_key.key]
  key_id = aws_kms_key.key.arn
}

output "public_key" {
  value = {
    public_key = data.aws_kms_public_key.key.public_key
    algs = data.aws_kms_public_key.key.signing_algorithms
    usage = data.aws_kms_public_key.key.key_usage
  }
}
```

### Expected Behavior

According to the [docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/kms_public_key#public_key) the `public_key` attribute should have returned the base64 encoded DER public key as received in the `PublicKey` attribute returned by the API call. 

### Actual Behavior

The `public_key` attribute returns a binary value, which doesn't even seem to be the raw DER, as if I `base64encode` it, it still isn't the correct value :-( 

I have compared with the output of `aws kms get-public-key --key-id <key-id>` and by viewing the public key in the AWS console.

All other attributes returned by this data source seem to be correct.

### Steps to Reproduce

1. `terraform apply`

### PR DESCRIPTION
**Pull request by cmeichel-treezor**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #20595.

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

**Review by ajegu-treezor — APPROVED** [review]

**Review by landru29 — APPROVED** [review]

File: aws/data_source_aws_kms_public_key.go [hunk 1]
```diff
@@ -75,7 +76,7 @@ func dataSourceAwsKmsPublicKeyRead(d *schema.ResourceData, meta interface{}) err
 	d.Set("arn", output.KeyId)
 	d.Set("customer_master_key_spec", output.CustomerMasterKeySpec)
 	d.Set("key_usage", output.KeyUsage)
-	d.Set("public_key", string(output.PublicKey))
+	d.Set("public_key", base64.StdEncoding.EncodeToString(output.PublicKey))
```
Comment (chrisclayson) [inline aws/data_source_aws_kms_public_key.go:79]: Is this correct? If my understanding is correct `output.PublicKey` is already base64 encoded, so should it be returned as is?

Comment (anGie44) [reply aws/data_source_aws_kms_public_key.go:79]: I think you're correct @chrisclayson -- from the AWS Go SDK Documentation https://docs.aws.amazon.com/sdk-for-go/api/service/kms/#GetPublicKeyOutput
> PublicKey is automatically base64 encoded/decoded by the SDK

Comment (chrisclayson) [reply aws/data_source_aws_kms_public_key.go:79]: Ah ... OK, fair enough, I was reading the AWS API docs, didn't realise the Go SDK automatically decoded it. Thanks for clarifying! 👍🏻

**Review by chrisclayson — APPROVED** [review]

**Review by anGie44 — APPROVED** [review]
Thanks for this fix @cmeichel-treezor , LGTM 🚀 

Output of acceptance test:
```
--- PASS: TestAccDataSourceAwsKmsPublicKey_basic (15.07s)
--- PASS: TestAccDataSourceAwsKmsPublicKey_encrypt (15.17s)
```

**Pull request merged**